crates/trycp_server/src/main.rs

```diff
@@ -291,6 +334,55 @@ fn main() {
     let players_arc_reset = players_arc.clone();
     let players_arc_spawn = players_arc;
 
+    // if we are acting as a manger add the "register" and "request" commands
+    if args.manager {
+        let state_registered = state.clone();
+        let state_request = state.clone();
+
+        // command for other trycp server to register themselves as available
+        io.add_method("register", |params: Params| {
+            let params_map = unwrap_params_map(params)?;
+            let url_str = get_as_string("url", &params_map)?;
+            let url = Url::parse(&url_str).map_err(|e| {
+                invalid_request(format!("unable to parse url:{} got error: {}", url_str, e))
+            })?;
+            let ram = get_as_int("ram", &params_map)?;
+
+            let mut state = state_registered.write().exect("should_lock");
+            state.registered.entry(url).or_insert(url) = NodeInfo{ram};
+            Ok(Value::String("registered".into()))
+        });
+
+        // command to request a list of available trycp_servers
+        io.add_method("request", |params: Params| {
+            let params_map = unwrap_params_map(params)?;
+            let count = get_as_int("count", &params_map)?;
+            let mut endpoints : Vec<Url,NodeInfo> = Vec::new();
+            let mut state = state_registered.write().exect("should_lock");
+
+            // confirm available nodes
+            if state.registered.len() >= count {
+                while count > 0 {
+                    let (url, info) = state.registered.pop()
+                    if check_url(url) {
+                        endpoints.push((url,info))
+                    }
+                    count -= 1;
+                }
+            }
+            if count > 0 {
+                // add any nodes that got taken off the registered list that are still valid back on
+                for (url,info) {
+                    state.registered.insedrt(url,info);
+                }
+                Ok(json!({ "error": "insufficient endpoints available" }))
+            }
+            else {
+                Ok(json!({ "endpoints": endpoints }))
+            }
+        });
+    }
```

Comment:
Seems like it would be nearly as easy to separate this out into its own binary. Is there a benefit to having this in the same binary as trycp_server? Seems totally orthogonal. But in any case if it's going to stay in the same binary, I would at least put an `else` block here and put all the other methods inside it, so that the thing acts as a manager XOR a server, never both.

Reply:
The main reason is not to have to add any other tooling/nix-shell commands etc.  I was also thinking that the manager could be one of the available trycp endpoints at the same time, which is why I didn't do that.  But maybe that's a bad idea?